src/tests/token/erc721/test_dual721.cairo

```diff
@@ -24,8 +24,8 @@ fn setup_snake() -> (DualCaseERC721, IERC721Dispatcher) {
     calldata.append_serde(OWNER());
     calldata.append_serde(TOKEN_ID);
 
-    start_cheat_caller_address(test_address(), OWNER());
     let target = utils::declare_and_deploy("SnakeERC721Mock", calldata);
+    start_cheat_caller_address(target, OWNER());
     (DualCaseERC721 { contract_address: target }, IERC721Dispatcher { contract_address: target })
```

Comment:
I'm not sure if this was addressed yet in guideline discussions, but it might be worth considering the pros and cons with setting a default caller in test setup functions. Maybe it's too granular, idk. Some of the tests needlessly set the caller to OWNER

Reply:
Let's refactor that in a separate PR, for the sake of wrapping up the foundry migration.